commons/ava_networks/test_gecko_network.go

```diff
@@ -176,16 +187,23 @@ func (loader TestGeckoNetworkLoader) ConfigureNetwork(builder *networks.ServiceN
 	// Add user-custom configs
 	for configId, configParams := range loader.serviceConfigs {
 		certProvider := cert_providers.NewRandomGeckoCertProvider(configParams.varyCerts)
+		alternateImageName := configParams.alternateImageName
 		initializerCore := ava_services.NewGeckoServiceInitializerCore(
-			loader.snowSampleSize,
-			loader.snowQuorumSize,
+			configParams.snowSampleSize,
+			configParams.snowQuorumSize,
 			loader.isStaking,
 			bootNodeIds,
 			certProvider,
 			configParams.serviceLogLevel)
 		availabilityCheckerCore := ava_services.GeckoServiceAvailabilityCheckerCore{}
-		if err := builder.AddTestImageConfiguration(configId, initializerCore, availabilityCheckerCore); err != nil {
-			return stacktrace.Propagate(err, "An error occurred adding Gecko node configuration with ID %v", configId)
+		if alternateImageName == nil {
+			if err := builder.AddTestImageConfiguration(configId, initializerCore, availabilityCheckerCore); err != nil {
```

Comment:
I'm not a fan of nil-as-optional when we can avoid it. Rather than doing this, let's just parameterize the whole TestSuite with `testImageName` and `byzantineImageName`, make every `TestGeckoNetworkServiceConfig` take in the image name (so the writer of the test has to think what they want, which is fair), and then this code here just uses whatever image is declared on the `TestGeckoNetworkServiceConfig`